### ISSUE TITLE
[Feature request] Option to automatically remove or rename remote mount points at unmount

### ISSUE THREAD

**Issue by Kabouik** [jarun/nnn#637]
**tl;dr** the current remote system can be error prone, more so than GUI alternatives. We all agree that the `nnn` workflow is by far more efficient, and with great powers comes great responsibility, but maybe there could be simple things to reduce the risk without changing too much the `nnn` rationale.

---

**What are the risks?**

`nnn` leaves empty folders after unmounting remotes. For some, this causes an urge to manually clean afterwards. I can see how some people would triple check that the folder is really empty before doing so, in fear of running `rm -rf` on the remote filesystem. There are a few reasons one could be in doubt, at least while still learning `nnn`, that don't really apply to GUIs: 
1. Does the mount persist after quitting `nnn`?
2. Is it tied to the saved session?
3. Is the connection automatically reinitialized after some inactive time if I enter the folder to double check that it's empty, and then when I'm back at the higher level to delete it, is my phone actually being remounted?
4. What happens if the wireless connection is dodgy and you see nothing in the folder in `nnn` while it tries to fetch the content (I know it happens often in GUI file managers, so that got me a little scared here)? I am used to `nnn` being lightning fast, so if I see no content in 0.5 second, I might concluded there's nothing. [Edit] I'm not sure actually if `nnn` would enter the folder before fetching the content, so maybe this 4th point is a non-issue.

**How is it specific to `nnn`?**

The answer to above questions is often obvious, but any bad judgement when trying to guess might lead to catastrophic damage, and since empty mount points persist in `nnn`, it can be ambiguous here. This can't happen on GUI FMs because they typically use a bookmark system for remote machines. In Nemo, Nautilus and Caja, I believe the bookmark just moves from "Bookmarks" to "Active remotes" when it's active. Then, if the user unmounts it, it moves back to its original location and loses the active icon. There is no real folder leftover on the system, no urge to clean, and the bookmark location and icon make it clear whether there is something mounted or not.

**How is it different than other mount points?**

Remotes are typically mounted and unmounted several times a day, most often just for short durations just while transferring a few files, and in that regard they're pretty different than other, local, mount points. A dodgy connection could also make them cycle between active or inactive with no actual user action, or appear only temporarily empty if the connection is instable/slow. No visual indicator on such always dynamic mount points can be very error prone.

**Solutions**

Possible solutions, if the above is deemed relevant, could be to:
- Remove remote mount points immediately when the remote is unmounted; however that would require typing a new mountpoint name at every new connection.
- Rename remote mount points depending on whether there's an active connection or not, to make it immediately visible the content is sensible. Possible rename options could include `Phone/` *vs* `Phone.unmounted/` or `.unmounted/Phone`, `ActiveRemotes/Phone` *vs* `InactiverRemotes/Phone`, *etc.*
- Favour a real ookmark system when prompting for the name of the remote (which actually is the name of the mount point to be created): offer to use a preexisting bookmark from a list or create a new one, and then move empty mount points elsewhere when inactive.

**Describe alternatives you've considered**

- Deleting manually, but that is very error prone for remotes I think.
- Leaving empty folders, but not knowing at first glance if they're active is concerning.

**Additional context**

I'm sorry, this is a lengthy way to put it but I know it contradicts to some extent the `nnn` simplicity so I wanted to elaborate why I think it may be important still. But I'll understand if no one agrees.

**Comment by jarun** [comment]
> Deleting manually, but that is very error prone for remotes I think.

We document this behaviour clearly: " nnn doesn't delete the mount point on unmount to prevent accidental data loss. Please ensure the mount point is not mounted if you are deleting it manually."

We can't rely on return values from third party utilities and do this risk free. Also, why do you want to delete? We generally access remote mounts regularly.

**Comment by jarun** [comment]
I can see why leaving the directory can be an annoyance in case of archive mounts. So I think removing the directory is a good idea indeed.

So, we can:

- support unmount hovered and attempt to remove the mountpoint
- remove the support for mount hovered directory

**Comment by jarun** [comment]
Please test and confirm if the behaviour works as expected.

**Comment by Kabouik** [comment]
Thanks @jarun! Just before I try in real world, I notice before the change that unmounting my phone always returns "Unmount failed, try lazy?". I just want to make sure there's no risk that `nnn` would try to delete the folder before trying the fallback lazy unmounting solution when the first unmount fails.

**Comment by jarun** [comment]
> I just want to make sure there's no risk that nnn would try to delete the folder before trying the fallback lazy unmounting solution when the first unmount attempt fails.

If lazy fails, `nnn` doesn't try to delete. Also, `nnn` uses `rmdir()`, so if the dir is not empty, it is not deleted. From the man page:

    rmdir() deletes a directory, which must be empty.

**Comment by Kabouik** [comment]
Excellent, `rmdir` sounds safe indeed. I can confirm it works, thanks for being so quick to implement it!

Maybe echo some "Remote unmounted" in the terminal if the `unmount` and `rmdir()` word, so that the state is clear when the user quits `nnn`? I know I would like it but that doesn't mean it would be relevant for others, I'm just throwing the idea but it's fine either way.

**Comment by jarun** [comment]
You get a msg when it fails. Be happy with it. We don't want to waste 300 ms of your life seeing useless messages every time you unmount. At least I don't want to and can't add an option to disable it. ;)

**Comment by Kabouik** [comment]
No problem, thanks!

**Comment by leovilok** [comment]
@Kabouik you ran use `rmdir` instead of `rm` to be sure to avoir recursively removing anything: <kbd>o</kbd>`rmdir`<kbd>Return</kbd><kbd>c</kbd>

**Comment by jarun** [comment]
@leovilok I have pushed the change to use `rmdir()` when unmount is successful. So no need to remove manually anymore.

**Comment by leovilok** [comment]
Sorry, I missed a few messages ...

**Comment by Kabouik** [comment]
Speaking of this, would it make sense to create the mount points in a directory that is **not** tied to `nnn`? I can understand why it was decided to put stuff mounted by `nnn` would be in the `nnn` folder. But on the other hand, nesting a remote file system inside a config folder (which people may manipulate outside `nnn` when forking/updating/synchronizing configs among their PCs) may be risky, especially as mounted remotes persist whether `nnn` is running or not.

For instance I wanted to fork `nnn` a few minutes ago to send a PR, and Iealized at the last minue when I was about to create a new fresh config folder that I had mounted my phone 3 hours ago. I was not so far from deleting the folder manually because I don't expect file systems here.

How about `/media/username/`?

**Comment by jarun** [comment]
We can definitely consider it as an improvement. I kept it the same as the archives. The problem is not all systems follow the /media hierarchy... I remember earlier RedHat used to have /mnt... How about macOS and *BSD?

**Comment by leovilok** [comment]
I's seen `/run/mount/user` too, but I think that putting mounts somewhere else would be a good idea, we use `${XDG_CONFIG_HOME}` for stuff that totally aren't [configuration files](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).

**Comment by Kabouik** [comment]
Instead of asking just the name of the remote when hitting `c` then `s`, we could ask the user to fill in the absolute path he wants, and possibly pre-fill `/media/username` or  `/run/media/`?

**Comment by jarun** [comment]
There is no standard directory and `nnn` config dir is already under `${XDG_CONFIG_HOME}`. I think we are OK with what we have right now. If people are not able to use `nnn` because they don't read/understand docs they should really use some other file manager that suits them. We are past the point where we spend time on issues like "I can't remember..." and "I missed...".

> we could ask the user to fill in the absolute path he wants, and possibly pre-fill /media/username or /run/media/

That just lengthens the workflow and makes the experience a drag. Note that wherever we mount, we take the user into it. Read my notes below.

> Is there a way to autodetect what is the system standard location for mounted media?

Not without messing with dev-rules. I do NOT want to touch them or read config files. We mount at a fixed location. That should suffice.

@Kabouik the above is the general stand from my side... nothing specific to your points. I consider my time more important than worth spending on random user usability issues.

Notes: When I mount a directory and the file manager takes me within it, the least expected from me is to remember unmounting it when I a done. Personally, I prefer to use a new context to copy files to/from the mounpoint so I don't have to leave it. When I'm done I trackback tot he mountpoint and press `u`. Maybe it's just a matter of standardizing your workflows.

**Comment by jarun** [comment]
Also I missed something - we are discussing remote mounts. The mention of `/media` distracted me into considering drives.

Remotes are not mounted automatically. If you remember to mount a remote then you should also remember to unmount it.

> For instance I wanted to fork nnn a few minutes ago to send a PR

> I don't expect mounted file systems to be tied to any software's own folder

I do not expect PRs from anyone so frequently that they should be deleting the `nnn` config dir every day.

**Comment by Kabouik** [comment]
> There is no standard directory and `nnn` config dir is already under `${XDG_CONFIG_HOME}`. I think we are OK with what we have right now. […] We are past the point where we spend time on issues like "I can't remember..." and "I missed...".
> […]
> @Kabouik the above is the general stand from my side... nothing specific to your points. I consider my time more important than worth spending on random user usability issues.

There is absolutely no problem with that and I totally respect that. I swear I read the docs and am trying to master `nnn` by myself, but I think the topic here is not only a matter of user workflow or environment issue.

There's a change of stance between your last and the one two hours ago. I don't know if this is my latest suggestion that changed your mind but that was just me throwing an idea to solve something you seemed to agree could be improved. There are probably better solutions. I just think having filesystems mounted in config directories is unusual, in a way that could cause issues because it's not standard expectation to have anything more sensible than configs there. It's not uncommon to use `rm` or `cp` in `~/.config/` to start from scratch or make modifications, and it's not uncommon either to mount remote systems and keep them mounted for later, there are use cases for that. But both situations don't go very well together.

Another way could be to have a `NNN_MOUNTDIR` environment variable and the default value that you think is best. But I do understand if you are reluctant and believe it is not desirable. It's fine by me, but I don't think it is a user preference or custom workflow issue, as pointed out by leovilok above.

> Notes: When I mount a directory and the file manager takes me within it, the least expected from me is to remember unmounting it when I a done. Personally, I prefer to use a new context to copy files to/from the mounpoint so I don't have to leave it. When I'm done I trackback to the mountpoint and press u. Maybe it's just a matter of standardizing your workflows.
I use contexts too, I find them very convenient for that. But sometimes I also have to leave `nnn`, or *want* to leave `nnn`, without necessarily wanting the remote to be unmounted. My belief was this shouldn't conflict with messing configs, but all this was just a discussion. You decide and I'm fine with it.

> I do not expect PRs from anyone so frequently that they should be deleting the nnn config dir every day. ;)
Okay. Well, that was just me trying to be helpful when you said that I should Google and PR. :/

**Comment by jarun** [comment]
> There's a change of stance between your last two messages and the one two hours ago.

They are two different issues. The first problem was about removing stale directories and I considered it a better approach because of stale archive mount dirs.

The second problem is unrelated to the first one where the mounted dir is not unmounted by user.

> I swear I read the docs and am trying to master nnn by myself

> it's not standard expectation to have anything more sensible than configs there.

The documentation is here: https://github.com/jarun/nnn/wiki/Basic-use-cases#sshfs

But I think it's a good idea to highlight at the top that `nnn` mounts withing the config directory.

To summarize: as there is no standard location for mounting remotes that's the best we can do. It's also convenient because hovering on the mountpoint and pressing `u` makes it simpler to unmount (and also remove the directory now).

I think the best option for you would be to add a prompt and provide the path manually every time you want to mount a remote instead of the current automation where you can just provide the remote name. I do not want the workflow because user has to type much more. And as there is no standard default, I do not want to add a config and stop it from working out of the box. I am sure someone else will miss that config and complain why it doesn't work out of the box.

**Comment by Kabouik** [comment]
Yes there were two issues (didn't feel like it was worth creating a new one without knowing what you'd think about it) but I thought your message two hours ago was about the location of mount points too. 

Yup I read that doc to get sshfs mount working, and it was clear to me when reading it that the mount point would be in `~/.config/nnn`, no surprise there.

The more I think about it, the more I think the best for me would be an additional NNN_MOUNTDIR environment variable (which, if unset, doesn't change the current behavior) to set the parent directory of all my mounts, and then retain the current behaviour of `nnn`, i.e., prompting for a remote name and mounting the remote in the corresponding new subfolder, but in `NNN_MOUNDIR/subfolder` instead of `XDG_CONFIG_HOME/subfolder`. I'll look into the code if I understand how to do that locally without breaking something else.

**Comment by leovilok** [comment]
> There is no standard directory and `nnn` config dir is already under ${XDG_CONFIG_HOME}. I think we are OK with what we have right now.

There is a lot of stuff in `.config/nnn` that is not configuration files. It's probably too late to change it, as `nnn` users are probably already used to that, but some files really belong to other [standards dirs](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

> $XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

This is where we should put sessions and plugins.

> $XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

I don't think that there is any configuration file *per se* in `nnn`, It might be the place to put shell files exporting env vars, to be sources in e.g. `.profile`.

> $XDG_CACHE_HOME defines the base directory relative to which user specific non-essential data files should be stored. If $XDG_CACHE_HOME is either not set or empty, a default equal to $HOME/.cache should be used.

It might be the place to put `.selection`? unless in:

> $XDG_RUNTIME_DIR defines the base directory relative to which user-specific non-essential runtime files and other file objects (such as sockets, named pipes, ...) should be stored.
...
> If $XDG_RUNTIME_DIR is not set applications should fall back to a replacement directory with similar capabilities and print a warning message. Applications should use this directory for communication and synchronization purposes and should not place larger files in it, since it might reside in runtime memory and cannot necessarily be swapped out to disk.

This is where we should have put everything we currently put in `/tmp` (which is incidently the common fallback dir for $XDG_RUNTIME_DIR).

Now this doesn't explicitely says where we should put mounted dirs, but following these definition, it pretty clear that  $XDG_CONFIG_HOME is not the right place, officially. Maybe `$XDG_RUNTIME_DIR` (fallback: /tmp) or `$XDG_DATA_HOME` (fallback: ~/.local/share)?

Anyways, as I said, it's probably too late to change everything to comply to standards.

**Comment by Kabouik** [comment]
Speaking of, although it might deviate a little from what all those folders should be used for, I must say that I personally like that everything `nnn`-related is in one single place; plugins, sessions, *etc.* Makes everything simpler for me. I just wanted to make that clear after the discussion above, the only thing I am more uncomfortable with (or fear) is invoking files that are not related to `nnn` in there. That being said, I'll see if I can patch `nnn` locally to make it do what I'd prefer.

**Comment by jarun** [comment]
> I'll look into the code if I understand how to do that locally without breaking something else.

:+1: That's exactly something I like people who stick with `nnn` do. I myself build `nnn` with 7/8 flags to misuse stuff the general public won't like.

> It's probably too late to change it

No, it's just too simple to ignore the benefits:

- one bookmark to reach all files related to `nnn`
- easy script for portability
- minimal checks when starting `nnn`
- lesser directories, lesser litter

Using multiple locations for multiple _types_ of files belonging to the same software is the worst of designs possible. I did not write the XDG-spec (?) and I do not think it has to be the gold standard. I like a simpler approach - binary in the PATH and a single directory where everything is well organized.

**Comment by Kabouik** [comment]
> +1 That's exactly something I like people who stick with `nnn` do. I myself build `nnn` with 7/8 flags to misuse stuff the general public won't like.

I'm not in the field of software development at all though, meaning there is a high risk of massive failure because I might just not understand the code.

I fully agree about the simplicity of having all files related to `nnn` in one single place.

**Comment by jarun** [comment]
> I'm not in the field of software development at all though

My very first contributions to open source began with this tiny and superfast distro. The person who helped me configure the Linux kernel SCSI related defaults for the next release was actually into dramas and plays. Then there was this guy from Oxford biotech who wrote a widely used kernel driver. Technology is something people should just learn as a hobby. Everyone doesn't have to be an engineer, they should just be creative.

**Comment by jarun** [comment]
And it also comes from instinct and need. Like how man learnt to build the fire. I was prepping a Pi 4 with Ubuntu 20.04 for my 6 yr old son last evening (while I was also responding here). He used to have it as a media center earlier. When I woke up this morning I found he has written his password (though it's very simple) in big letters on a slate.

So he has observed that the login screen retains the username but not the password and did the most sensible thing he could think to remember it. That's what I am talking about. You will do it when you need it badly enough. I never told him he needs to remember his password.

**Comment by Kabouik** [comment]
That's exactly what I'm trying to do on some other open source stuff lately, but I'm far trailing far behind you. Or your son. :<

For now, there are just two things I'd like to change in `nnn` in the future if I can understand enough from the code, and they're both for security reasons because, while I love the speed of nnn, I want to shift the cursor down a little if it can significantly reduce the risk of data loss due to human mistakes. So my to-do list is (1) use an environment variable to set a base mount dir outside `/.config/nnn` but keep the rest as is (automatically navigate to the mount point), and (2) restrict the filr/dir deletion action to a `^X` only and scrap `x` to limit the risk of accidental deletion. Probably not capable to do that right now, but hopefully I'll learn by keeping an eye on development.

My little victory of the day was trying [this](https://github.com/Calinou/0x0/tree/master/bin) as a plugin and seeing it works perfectly for both text and image uploads. Small things. But what is more significant is I can see myself getting faster and faster with using nnn already. And besides speed, the benefits in usability are already huge when I can use $EDITOR to edit text files in multiple locations so quickly without having to make a project in an IDE.

**Comment by jarun** [comment]
> Or your son.

LOL!

> So my to-do list is (1) use an environment variable to set a base mount dir outside /.config/nnn but keep the rest as is (automatically navigate to the mount point)

:+1: 

> restrict the filr/dir deletion action to a ^X only and scrap x to limit the risk of accidental deletion

:+1: There are contradictory opinions on this. Some people refuse to use Shift/Ctrl keys. Some say they find `x` risky. We have a confirmation in place.

It's very easy to change though. Here's the patch:

```diff
diff --git a/src/nnn.h b/src/nnn.h
index 1f7f664..58c8a63 100644
--- a/src/nnn.h
+++ b/src/nnn.h
@@ -222,7 +222,6 @@ static struct key bindings[] = {
        { 'w',            SEL_CPMVAS },
        { CONTROL('W'),   SEL_CPMVAS },
        /* Delete from selection buffer */
-       { 'x',            SEL_RM },
        { CONTROL('X'),   SEL_RM },
        /* Open in a custom application */
        { 'o',            SEL_OPENWITH },
```

**Comment by jarun** [comment]
> My little victory of the day was trying this as a plugin and seeing it works perfectly for both text and image uploads.

Can you take a look at the `upload` plugin? This looks to be int he same lines.

> But what is more significant is I can see myself getting faster and faster with using nnn already

Exactly. Keep reading the documentation as well. There are many many hidden things in `nnn`.

**Comment by leovilok** [comment]
> Using multiple locations for multiple types of files belonging to the same software is the worst of designs possible. I did not write the XDG-spec (?) and I do not think it has to be the gold standard. I like a simpler approach - binary in the PATH and a single directory where everything is well organized.

The unique dir way of `nnn` makes sense, and it's kind of nice, but XDG specs totally make sens too. I think that the rationale is something like that:
- you're supposed to be able to copy/backup or even share configuration files, so if every application follow the standard, you can copy the whole `.config/` dir to other machines, or even make it a git repo
- having user data files in a directory allows backing them up
- you can put cache & temporary user data files in RAM filesystems, so that every write is not a disk write (I believe users compained about that in `nnn`)

`nnn` put a lot of files of different types in the same dir, which makes it hard to do any of this:
- it's useless to backup selection, and you definitively want to avoid backing up mount contents
- we might want to put selection (often written) in ramfs, but plugins & sessions must be on a persistent FS
...

**Comment by jarun** [comment]
We have NNN_SEL to put selection anywhere.

**Comment by jarun** [comment]
> you can copy the whole .config/ dir to other machines

Speaking of `nnn`, all the files in .config/nnn are supposed to be copied to the new machine.

Also, I find it interesting that people believe that a system can be cloned manually, application by application (if you want ALL config files, definitely you want ALL apps too). The sanest way to _clone_ a system is to take a bare metal/OS backup and deploy it in a new system. Where everything goes.

No I wouldn't share all my config files with anyone risking secrets and keys littered here and there.

Somehow the XDG advocates have linked into a lot of unrelated and fancy use cases which were always possible even when XDG was not in place.

> having user data files in a directory allows backing them up

You can backup any permanent directory anywhere in the fs. Every backup service supports cherry-picking.

> you can put cache & temporary user data files in RAM filesystems, so that every write is not a disk write (I believe users compained about that in nnn)

Agreed. And we have NNN_SEL to put selection anywhere.

**Comment by jarun** [comment]
Anyway, @leovilok I think we have a middle ground. We do not have a config file and .selection can be configured by `NNN_SEL`. How about moving everything to `XDG_DATA_HOME` and be happy about it?

Feel free to raise a PR.

Note that the first check would be if we need to migrate the current files (maybe the `nnn` directory) safely if it exists and move to `XDG_DATA_HOME`.

If not, then setup as usual with the location changed.

**Comment by Kabouik** [comment]
> It's very easy to change though. Here's the patch:

Awesome, I'll try that. I was looking in `nnn.c`, turns out that was in `nnn.h`. I just to adjust my update workflow to apply this patch every time I update `nnn`. Today I managed to write my very complicated plugin to use in conjunction with `trash-cli` on `^X`, forceful removal function like default, but with conservative confirmation and not directly accessible:
```
#!/usr/bin/env sh

# Description: remove with `rm`
# Meant to be used to bypass trash for those who enabled NNN_TRASH=1 for the default nnn behavior
#
# Shell: POSIX compliant
# Authors: Kabouik, matf@disr.it

rm -rI $1
```
More comments than code. :100: 

> Can you take a look at the upload plugin? This looks to be in the same lines.

Absolutely, it's very similar, but I like 0x0 more and it uploads images and returns a direct URL with no html decorations. That can probably be done too with minimal efforts with `upload`, or I could just change the domain; same with `imgur` I suppose.

Regarding `NNN_MOUNTDIR`:

```diff
diff --git a/src/nnn.c b/src/nnn.c
index e822a4c..25f01ef 100644
--- a/src/nnn.c
+++ b/src/nnn.c
@@ -3904,7 +3904,7 @@ static bool archive_mount(char *newpath)
        DPRINTF_S(dir);
 
        /* Create the mount point */
-       mkpath(cfgdir, dir, newpath);
+       mkpath("NNN_MOUNTDIR", dir, newpath);
        free(dir);
 
        if (!xmktree(newpath, TRUE)) {
@@ -3957,7 +3957,7 @@ static bool remote_mount(char *newpath)
        }
 
        /* Create the mount point */
-       mkpath(cfgdir, tmp, newpath);
+       mkpath("NNN_MOUNTDIR", tmp, newpath);
        if (!xmktree(newpath, TRUE)) {
                printwarn(NULL);
                return FALSE;
@@ -4013,7 +4013,7 @@ static bool unmount(char *name, char *newpath, int *presel, char *currentpath)
        }
 #endif
 
-       if (tmp && strcmp(cfgdir, currentpath) == 0) {
+       if (tmp && strcmp("NNN_MOUNTDIR", currentpath) == 0) {
                mkpath(cfgdir, tmp, newpath);
                child = lstat(newpath, &sb) != -1;
                parent = lstat(xdirname(newpath), &psb) != -1;
@@ -4031,7 +4031,7 @@ static bool unmount(char *name, char *newpath, int *presel, char *currentpath)
        }
 
        /* Create the mount point */
-       mkpath(cfgdir, tmp, newpath);
+       mkpath("NNN_MOUNTDIR", tmp, newpath);
        if (!xdiraccess(newpath)) {
                *presel = MSGWAIT;
                return FALSE;
```

Would that be enough or would it break things?

**Comment by leovilok** [comment]
@jarun yeah, it's too late, It would mean rewriting a lot of stuff for something that works already an is used by a lot of people.

If it starts generating issues for actual users, we could have a `NNN_DIR` variable, defaultingt to the current location. But I think that there is no need to fix it while it's not broken. And we are not alone "abusing" the config dir to put non config data, so I guess we won't be the one project to break users' configuration.

I still don't really like to put mounted dirs right next to other stuff, maybe I'd like it better if it was either a subdir, or configureable with something like `NNN_MOUNTDIR` ... but it doesn't bother me enough to change anything, I can totally live with it. We're all good :+1:

**Comment by jarun** [comment]
> better if it was either a subdir

I think we can have a `mounts` dir. I'll implement that.

**Comment by jarun** [comment]
Subdir "mounts" is implemented at commit b62adec72ce31918e2ccd903bd29d382b03b799d.

**Comment by leovilok** [comment]
Nice!

**Comment by Kabouik** [comment]
> I myself build `nnn` with 7/8 flags to misuse stuff the general public won't like.

Would you mind explaining exactly what is your workflow for that? I started accumulating some custom changes locally (extra plugins, modified plugins, removal of the `x` to delete and `?` info edited accordingly, _etc._) but i don't want to miss `nnn` updates because there's just too much cool stuff at each update. 

So far I `git stash`, `git pull`, `git stash apply`, `sudo make strip install`, but I was wondering if there was not a better workflow to this. I assume many people face the same need when dealing with  plugins for instance. Or does everyone use a different folder for git and for their actual `nnn` files?

**Comment by jarun** [comment]
> Would you mind explaining exactly what is your workflow for that?

I meant make options: https://github.com/jarun/nnn/wiki/Developer-guides#make-options

I use the most trimmed down version of `nnn` with the following resource usage:

```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                
 1811 vaio      20   0   11932   2848   2012 S   0.0  0.0   0:00.03 nnn -cEFnQrSx -t 900
```
